sdk/js/index.ts

```diff
@@ -272,32 +285,44 @@ export default class StreamManager {
   }
 
   onStreamCreated(
-    handleStream: (stream: Stream) => null,
+    handleStream: (stream: Stream) => void,
     creator?: Address
   ): void {
-    const onLogs = (logs: Log[]) => {
-      logs
-        .map(
-          // Log is StreamCreated
-          (log: Log) =>
-            new Stream(
-              this,
-              log.topics[2] as Address, // creator
-              Number(log.topics[3]), // streamId
-              log.topics[4] as Address, // token
-              BigInt(log.topics[5] as string), //amount per second
-              this.publicClient,
-              this.walletClient
-            )
-        )
-        .forEach(handleStream);
-    };
     this.publicClient.watchContractEvent({
       address: this.address,
       abi: StreamManagerContractType.abi as Abi,
       eventName: "StreamCreated",
       args: creator ? { creator } : {},
-      onLogs,
+      onLogs: (logs: StreamCreated[]) => {
+        logs
+          .map((log) => {
+            return new Stream(
+              this,
+              log.args.creator,
+              log.args.stream_id as number,
+              log.args.token,
+              BigInt(log.args.amount_per_second),
+              this.publicClient,
+              this.walletClient
+            );
+          })
+          .forEach(handleStream);
+      },
+      onError: (error) => console.log(error),
     });
   }
+
+  async fetchAllLogs(callback: (logs: Log[]) => void, fromBlock?: bigint) {
```

Comment:
We can probably switch this into the promise API using the handler function so it doesn't need to be async (since it doesn't return anything)